testplan/runnable/interactive/base.py

```diff
@@ -157,10 +157,8 @@ def run_all_tests(self, runner_uid=None, await_results=True):
 
         all_tests = self.all_tests(runner_uid)
 
-        return self._create_testplan_report(
+        for test_uid, real_runner_uid in all_tests:
             self.run_test(test_uid, real_runner_uid)
```

Comment:
If you don't want to return a list of [future](https://github.com/Morgan-Stanley/testplan/blob/4804ec125dab9adc3f8d056c4d27ad29faec273b/testplan/runnable/interactive/base.py#L734) perhaps `yield self.run_test(test_uid, real_runner_uid)`? 

I'm not stoked on ignoring the return value if my understanding is correct:
- A [concurrent.futures.Future](https://docs.python.org/3/library/concurrent.futures.html?highlight=concurrent%20futures#concurrent.futures.Future) is returned from [testplan.runnable.interactive.base.TestRunnerIHandler._run_async](https://github.com/ryan-collingham/testplan/blob/4804ec125dab9adc3f8d056c4d27ad29faec273b/testplan/runnable/interactive/base.py#L726)
 - This `Future` is _maybe_ returned from [testplan.runnable.interactive.base.TestRunnerIHandler.run_test](https://github.com/ryan-collingham/testplan/blob/4804ec125dab9adc3f8d056c4d27ad29faec273b/testplan/runnable/interactive/base.py#L163) (which I have other comments about, see below)
- If `run_test` _does_ return this `Future`, it's ignored here.

Reply:
No, you've misunderstood I think.

If run_all_tests is called with await_results=False, then a single future object is returned (which represents the eventual result of running all the tests)

If await_results=True (the default), then nothing is returned - the report object is updated as a side-effect. run_test() will _not_ return a future object, since it is implicitly passed await_results=True (as the default value for that param). So there is never any future object to ignore.

Does that make sense?

Reply:
All perfectly clear.

[The previous version](https://github.com/Morgan-Stanley/testplan/blob/15abdad4670808c6054e385c6e628f16b80197f6/testplan/runnable/interactive/base.py#L160) returns a report object or a Future.

[Your version](https://github.com/Morgan-Stanley/testplan/blob/4804ec125dab9adc3f8d056c4d27ad29faec273b/testplan/runnable/interactive/base.py#L160) returns nothing or a Future that resolves to None.

The question posed is, why not just return the report or a Future that resolves to the report.

Reply:
This is all relatively new code (this method added two months ago) and its usage throughout testplan is confined only to the interactive runner, which no other part of testplan depends on and which no client is using. So, my concerns about API breakage are moot.